lib/metadataTypes/MetadataType.js

```diff
@@ -1334,30 +1283,28 @@ class MetadataType {
     }
     /**
      *
-     * @param {Object} response response payload from REST API
+     * @param {Object} ex response payload from REST API
      * @returns {void}
      */
-    static checkForErrors(response) {
-        if (response && response.res.statusCode >= 400 && response.res.statusCode < 600) {
+    static checkForErrors(ex) {
+        if (ex && ex.response && ex.response.status >= 400 && ex.response.status < 600) {
             const errors = [];
-            if (response.body.errors) {
-                for (const errMsg of response.body.errors) {
+            if (ex.response.data.errors) {
+                for (const errMsg of ex.response.data.errors) {
                     errors.push(errMsg.message.split('<br />').join(''));
                 }
-            } else if (response.body.validationErrors) {
-                for (const errMsg of response.body.validationErrors) {
+            } else if (ex.response.data.validationErrors) {
+                for (const errMsg of ex.response.data.validationErrors) {
                     errors.push(errMsg.message.split('<br />').join(''));
                 }
-            } else if (response.body.message) {
-                errors.push(response.body.message);
+            } else if (ex.response.data.message) {
+                errors.push(ex.response.data.message);
             } else {
-                errors.push(`Undefined Errors: ${JSON.stringify(response.body)}`);
+                errors.push(`Undefined Errors: ${JSON.stringify(ex.response.data)}`);
             }
-            throw new Error(
-                `Errors on upserting metadata at ${response.res.request.path}: ${errors.join(
-                    '<br />'
-                )}`
-            );
+            Util.logger.debug(JSON.stringify(ex.config));
+            Util.logger.debug(JSON.stringify(ex.response.data));
+            return `Errors on upserting metadata at ${ex.request.path}: ${errors.join('<br />')}`;
```

Comment:
```suggestion
            Util.logger.debug(`Errors on upserting metadata at ${ex.request.path}`);
            return `\n- ${errors.join('\n- ')}`;
```
this shouldn't use HTML since it will be printed to error log files & CLI
also, no intro msg needed because the errors will be logged together with more details outside of this method. hence we should limit the return value to a minimum here

- [ ] keep return value clean

Reply:
push @DougMidgley